ibis/spark/tests/test_ddl_compilation.py

```diff
@@ -0,0 +1,245 @@
+
+
+@pytest.fixture
+def expr(t):
+    return t[t.bigint_col > 0]
+
+
+def test_create_external_table_as(mockcon):
+    path = '/path/to/table'
+    select, _ = _get_select(
+        mockcon.table('test1'), SparkDialect.make_context()
+    )
+    statement = ddl.CTAS(
+        'another_table',
+        select,
+        can_exist=False,
+        path=path,
+        database='foo',
+    )
+    result = statement.compile()
+
+    expected = """\
+CREATE TABLE foo.`another_table`
+USING PARQUET
+LOCATION '{0}'
+AS
+SELECT *
+FROM test1""".format(
+        path
+    )
+    assert result == expected
+
+
+def test_create_table_with_location_compile():
+    path = '/path/to/table'
+    schema = ibis.schema(
+        [('foo', 'string'), ('bar', 'int8'), ('baz', 'int16')]
+    )
+    statement = ddl.CreateTableWithSchema(
+        'another_table',
+        schema,
+        can_exist=False,
+        format='parquet',
+        path=path,
+        database='foo',
+    )
+    result = statement.compile()
+
+    expected = """\
+CREATE TABLE foo.`another_table`
+(`foo` string,
+ `bar` tinyint,
+ `baz` smallint)
+USING PARQUET
+LOCATION '{0}'""".format(
+        path
+    )
+    assert result == expected
+
+
+def test_create_table_parquet_with_schema():
+    directory = '/path/to/'
+
+    schema = ibis.schema(
+        [('foo', 'string'), ('bar', 'int8'), ('baz', 'int16')]
+    )
+
+    statement = ddl.CreateTableWithSchema(
+        'new_table',
+        schema,
+        database='foo',
+        format='parquet',
+        can_exist=True,
+        path=directory,
+    )
+
+    result = statement.compile()
+    expected = """\
+CREATE TABLE IF NOT EXISTS foo.`new_table`
+(`foo` string,
+ `bar` tinyint,
+ `baz` smallint)
+USING PARQUET
+LOCATION '{0}'""".format(
+        directory
+    )
+
+    assert result == expected
+
+
+def test_create_table_parquet(expr):
+    statement = _create_table(
+        'some_table', expr, database='bar', can_exist=False
+    )
+    result = statement.compile()
+
+    expected = """\
+CREATE TABLE bar.`some_table`
+USING PARQUET
+AS
+SELECT *
+FROM functional_alltypes
+WHERE `bigint_col` > 0"""
+    assert result == expected
+
+
+def test_no_overwrite(expr):
+    statement = _create_table('tname', expr, can_exist=True)
+    result = statement.compile()
+
+    expected = """\
+CREATE TABLE IF NOT EXISTS `tname`
+USING PARQUET
+AS
+SELECT *
+FROM functional_alltypes
+WHERE `bigint_col` > 0"""
+    assert result == expected
+
+
+@pytest.mark.xfail(raises=AssertionError, reason='NYT')
+def test_partition_by():
+    assert False
+
+
+def _create_table(
+    table_name, expr, database=None, can_exist=False, format='parquet'
+):
+    ast = build_ast(expr, SparkDialect.make_context())
+    select = ast.queries[0]
+    statement = ddl.CTAS(
+        table_name,
+        select,
+        database=database,
+        format=format,
+        can_exist=can_exist,
+    )
+    return statement
+
+
+def _get_select(expr, context):
```

Comment:
The context is never used where this function is called, so do we really need this function? It's not clear to me what's being saved here.